### PR TITLE
Config refactor

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Move config loading to loadConf:
Prevent hardcoding paths in multiple files
Allow using local configuration in dev mode and /etc/nimha for production
Bugfix: create certificates table at start time

### CODE REVIEW

File: nimha.nim
Comment (ThomasTJdev) [inline nimha.nim]: Why remove the check for mosquitto?

File: nimhapkg/resources/utils/common.nim
Comment (ThomasTJdev) [inline nimhapkg/resources/utils/common.nim]: `/etc/nimha/nimha.cfg` does not exists. It needs to created.

If we are going to move the `config_default.cfg` to both `getAppDir() & "/config/secret.cfg"` and `/etc/nimha/nimha.cfg`, we should inform the user that he needs to update the config in 2 places. Or an implementation where the user could edit the config in the browser, where the backend updates 2 places; or a simple symlink.

https://github.com/ThomasTJdev/nim_homeassistant/blob/2c28eab8e1c152d20b64074f00ab437e2d58422c/nimha.nim#L38

File: nimhapkg/resources/utils/common.nim
Comment (ThomasTJdev) [inline nimhapkg/resources/utils/common.nim]: `/etc/nimha/" & modulename & ".cfg` does not exists. Are you planning to split the *.cfg-file per module or what are your thoughts on this approach? If the file is split into multiple files, it could cause unnecessary work for the user, who needs to open and close a file per module (e.g. when we support +100 plugins, it would take a considerable amount of time).

File: nimha.nim
Comment (ThomasTJdev) [inline nimha.nim:132]: Good catch! We could delete the `certDatabase(dbWeb)` from `web_utils.nim` then.

https://github.com/ThomasTJdev/nim_homeassistant/blob/e23df125bb2dd8b7b76b6c3b904c7e040f7442c4/nimhapkg/modules/web/web_utils.nim#L25

**Review by ThomasTJdev — CHANGES_REQUESTED** [review]
Needs to be able to run with both `-d:dev` and `-d:release`. `/etc/nimha/nimha.cfg` is currently failing.

File: nimhapkg/resources/utils/common.nim
Comment (FedericoCeratto) [reply nimhapkg/resources/utils/common.nim]: Configs under /etc must be treated as read-only by applications. If you want to edit them from the browser we can switch to /var/lib/nimha/config/

File: nimhapkg/resources/utils/common.nim
Comment (FedericoCeratto) [reply nimhapkg/resources/utils/common.nim]: The main reason is that one module should not be able to access credentials stored in a config file for another module. On second thought everything could stay in one config file and then nimha coordinates access (e.g. hands the right config chunk to the right module).

File: nimhapkg/resources/utils/common.nim
Comment (ThomasTJdev) [reply nimhapkg/resources/utils/common.nim]: /etc would be fine - the current options in the config file should be more or less static values. The different modules keeps their config in databases.

But I still think, that we should help or inform the user to move the `/config/secret.cfg` to `/etc/nimha/nimha.cfg`. Currently we generate the raw `secret.cfg` in `proc secretCfg()`, but that'll only work for `-d:dev`. We should support an plug-and-play solution.

Info: the name `secret.cfg` is derived from Home Assistant. We can change it to nimha.cfg if you like.

File: nimhapkg/resources/utils/common.nim
Comment (ThomasTJdev) [reply nimhapkg/resources/utils/common.nim]: Good thought on not sharing credentials, that could be important for future modules. On the other hand, if you install a module, I would expect the user to trust the module code for not accessing other stuff without reason. I think one config file would give the best UX when editing them, but we could simulate that through editing them via the browser. I'm open for both suggestions.

File: nimha.nim
Comment (FedericoCeratto) [reply nimha.nim]: Fixed

File: nimhapkg/resources/utils/common.nim
Comment (FedericoCeratto) [reply nimhapkg/resources/utils/common.nim]: I switched back to a single file (at least temporarily) so that the PR can be merged. I'll open more PRs.

**Comment by ThomasTJdev** [comment]
Hi @FedericoCeratto 

## Nimble install
When NimHA is installed with `nimble` the first run will fail due to the config is missing in `/etc/nimha/`. We should inform the user to move the `secret_default.cfg`. `nimha.nimble` needs to be updated with this information before merge. 
**Example:**
```nim
echo "nimha.cfg: Please copy config/secret_default.cfg to /etc/nimha/nimha.cfg to run in -d:release mode (folder must be created). \n" 
```
**Reference:**
https://github.com/ThomasTJdev/nim_homeassistant/blob/fbdb548f7de1d5f49f641562973bf7a3b1d59191/nimha.nimble#L34-L35


### Config filename
We could rename the `secret_config.cfg` to `nimha_dev.cfg` to have consistency.


## Commits to PR
Would it be ok, if I push small changes like the above to your PR? Or would you prefer to manage the commits by yourself?

**Comment by FedericoCeratto** [comment]
Do you expect people to install nimha with nimble for production use? It's a little bit odd because nimble is not a system installer and this would require more glue to install systemd unit files, create /etc/nimha, create the system user and so on.

Commits to PR: PRs are piling up and preventing me from adding more stuff without messy rebases. What about adding your commits and then merging? I'll use a single PR in future.

**Comment by ThomasTJdev** [comment]
> Do you expect people to install nimha with nimble for production use? It's a little bit odd because nimble is not a system installer and this would require more glue to install systemd unit files, create /etc/nimha, create the system user and so on.

Yes. All previously versions has been tested to work out of the box with nimble. I have no clue, if anyone has used nimble with it - I personally prefer to clone and compile it.

> Commits to PR: PRs are piling up and preventing me from adding more stuff without messy rebases. What about adding your commits and then merging? I'll use a single PR in future.

I pushed a [PR](https://github.com/FedericoCeratto/nim_homeassistant/pull/1) since I dont have write access to this PR.

**Comment by FedericoCeratto** [comment]
Odd, ' Allow edits from maintainers.' was set.
I'll add files to do package-based installs and also a script to install after doing a nimble build.

**Pull request merged**
